Add new Prometheus config page, update _index file (#70272)
grafana/grafana: docs/sources/datasources/prometheus/_index.md

# Prometheus data source

> Prometheus is an open-source toolkit that collects and stores metrics used for monitoring and alerting. If you are just getting started with Prometheus, see [What is Prometheus?]({{< relref "../../fundamentals/intro-to-prometheus/" >}}).

Reviewer: ```suggestion
Prometheus is an open-source database that uses an telemetry collector agent to scrape and store metrics used for monitoring and alerting. If you are just getting started with Prometheus, see [What is Prometheus?]({{< relref "../../fundamentals/intro-to-prometheus/" >}}).
```

> Prometheus is an open-source toolkit that collects and stores metrics used for monitoring and alerting. If you are just getting started with Prometheus, see [What is Prometheus?](https://grafana.com/docs/grafana/next/fundamentals/intro-to-prometheus/)

Reviewer 2: This should be a relref.

Author: I am inclined to leave as is since we are looking at changing how links are handled...is that fair?

Author: changed to rel ref

If you run Grafana in an Amazon EKS cluster, follow the AWS guide to [Query using Grafana running in an Amazon EKS cluster](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query-grafana-7.3.html).

## Exemplars

> Exemplars are unique to Prometheus. Exemplars associate higher-cardinality metadata from a specific event with traditional time series data. See [Introduction to exemplars](https://grafana.com/docs/grafana/latest/fundamentals/exemplars/) in Prometheus documentation for detailed information on how they work.

Reviewer 2: change to be a relref

Author: changed to relref

{{% admonition type="note" %}}
Available in Prometheus v2.26 and higher with Grafana v7.4 and higher.
{{% /admonition %}}

> Grafana 7.4 and higher can show exemplars data alongside a metric both in Explore and in Dashboards.

Reviewer 2: We version our docs. We don't need to say this.

Author: leaving as is for now - I don't think the versioning here is misleading

> {{< figure src="/static/img/docs/v74/exemplars.png" class="docs-image--no-shadow" caption="Screenshot showing the detail window of an Exemplar" >}}

Reviewer 2: Is this screenshot still accurate? I see v7.4 in the path so it's pretty old.

Author: still accurate

To configure exemplars see [Exemplars](/docs/grafana/latest/datasources/prometheus/configure-prometheus-data-source/#exemplars).

<!-- Configure Exemplars in the data source settings by adding external or internal links. -->

> {{< figure src="/static/img/docs/prometheus/exemplars-10-1.png" max-width="500px" class="docs-image--no-shadow" caption="Exemplars" >}}

Reviewer 2: What is this image showing? Should there be an intro to the image, like "The following image shows..."?

> ## Incremental Dashboard Queries (beta)

Reviewer 2: should be sentence case

> As of Grafana 10, the Prometheus data source can be configured to query live dashboards incrementally, instead of re-querying the entire duration on each dashboard refresh.

Reviewer 2: we don't need to mention versions in our docs.

Author: I am leaving since it's new in the current version

This can be toggled on or off in the data source configuration or provisioning file (under `incrementalQuerying` in jsonData).

> Additionally, the amount of overlap between incremental queries can be configured using the `incrementalQueryOverlapWindow` jsonData field, the default value is 10m (10 minutes).

Reviewer 2: ```suggestion
Additionally, the amount of overlap between incremental queries can be configured using the `incrementalQueryOverlapWindow` jsonData field, the default value is `10m` (10 minutes).
```

If you run Grafana in an Amazon EKS cluster, follow the AWS guide to [Query using Grafana running in an Amazon EKS cluster](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query-grafana-7.3.html).

## Exemplars

> Exemplars are unique to Prometheus. Exemplars associate higher-cardinality metadata from a specific event with traditional time series data. See [Introduction to exemplars](https://grafana.com/docs/grafana/latest/fundamentals/exemplars/) in Prometheus documentation for detailed information on how they work.

Reviewer: I'm not sure this is technically correct. Exemplars might work with Mimir, and though this is just a gut suggestion, I think it's an important nuance to confirm.

Author: checking with Brendan

{{% admonition type="note" %}}
Available in Prometheus v2.26 and higher with Grafana v7.4 and higher.
{{% /admonition %}}

Grafana 7.4 and higher can show exemplars data alongside a metric both in Explore and in Dashboards.

{{< figure src="/static/img/docs/v74/exemplars.png" class="docs-image--no-shadow" caption="Screenshot showing the detail window of an Exemplar" >}}

> To configure exemplars see [Exemplars](/docs/grafana/latest/datasources/prometheus/configure-prometheus-data-source/#exemplars).

Reviewer 2: ```suggestion
To configure exemplars see [Exemplars](/docs/grafana/latest/datasources/prometheus/configure-prometheus-data-source#exemplars).
```
